code/game/objects/items/shields.dm

```diff
@@ -57,21 +61,17 @@
 		if(0 to 25)
 			. += span_warning("It's falling apart!")
 
-/obj/item/shield/proc/shatter(mob/living/carbon/human/owner)
-	playsound(owner, shield_break_sound, 50)
-	new shield_break_leftover(get_turf(src))
-
 /obj/item/shield/proc/on_shield_block(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", damage = 0, attack_type = MELEE_ATTACK, damage_type = BRUTE)
 	if(!breakable_by_damage || (damage_type != BRUTE && damage_type != BURN))
 		return TRUE
-	if (atom_integrity <= damage)
-		var/turf/owner_turf = get_turf(owner)
-		owner_turf.visible_message(span_warning("[hitby] destroys [src]!"))
-		shatter(owner)
-		qdel(src)
-		return FALSE
-	take_damage(damage)
-	return TRUE
+	take_damage(damage, damage_type, attack_type)
```

Comment:
Unfortunately, ``attack_type`` doesn't aligns with armor flag categories. I think you'll need to determine this separately. Since only projectiles have varying armor targets, we can just check the armor flag of the projectile if it is a ranged attack, and for anything else we can conveniently assume ``MELEE`` to be a workable default.

Reply:
Done that. I've also added armor penetration into account, at the cost of a few `else if` calls.